File: benchmarks/python/enums_match.py
from enum import Enum


class Mood(Enum):
    HAPPY = "happy"
    NEUTRAL = "neutral"
    GRUMPY = "grumpy"


def describe(m: Mood) -> str:
    if m is Mood.HAPPY:
        return "all sunshine"
    if m is Mood.NEUTRAL:
        return "perfectly balanced"
    return "approach with snacks"


def main() -> None:
    today = Mood.GRUMPY
    print(f"the cat is {today.value}")
    print(describe(today))
    if today is Mood.HAPPY:
        print("pet it")
    else:
        print("feed it first")


if __name__ == "__main__":
    main()

File: benchmarks/rust/enums_match.rs
use std::fmt;

#[allow(dead_code)]
enum Mood {
    Happy,
    Neutral,
    Grumpy,
}

impl fmt::Display for Mood {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Mood::Happy => write!(f, "happy"),
            Mood::Neutral => write!(f, "neutral"),
            Mood::Grumpy => write!(f, "grumpy"),
        }
    }
}

fn describe(m: &Mood) -> &'static str {
    match m {
        Mood::Happy => "all sunshine",
        Mood::Neutral => "perfectly balanced",
        Mood::Grumpy => "approach with snacks",
    }
}

fn main() {
    let today = Mood::Grumpy;
    println!("the cat is {}", today);
    println!("{}", describe(&today));
    match today {
        Mood::Happy => println!("pet it"),
        _ => println!("feed it first"),
    }
}
